### PR TITLE
adding total chassis power

### PR DESCRIPTION
**Pull request by ben-kulp**
**Summary**
- Adds collection of total GPU power consumption metric from Redfish Control endpoints
- Collects from /redfish/v1/Chassis/{chassisID}/Controls/TotalGPU_Power_0
- Uses gofish Control API to fetch sensor readings

**Implementation Details**
- Added chassis_gpu_total_power_watts metric to chassis collector
- Attempts collection for all chassis (gracefully handles 404s for chassis without the control)
- Validates control type is PowerControlType before emitting metric

**Testing**
- Added unit tests for single and multiple chassis configurations
- Added error handling tests for edge cases (zero readings, wrong control types)
- All existing tests continue to pass

```
# HELP redfish_chassis_gpu_total_power_watts Total GPU power consumption for all GPUs in chassis in watts
# TYPE redfish_chassis_gpu_total_power_watts gauge
redfish_chassis_gpu_total_power_watts{chassis_id="HGX_Chassis_0",resource="chassis"} 673.8720092773438
```

### CODE REVIEW

File: collector/chassis_collector.go [hunk 1]
```diff
@@ -521,3 +528,63 @@ func parseNetworkPort(ch chan<- prometheus.Metric, chassisID string, networkPort
 		ch <- prometheus.MustNewConstMetric(chassisMetrics["chassis_network_port_health_state"].desc, prometheus.GaugeValue, networkPortHealthStateValue, chassisNetworkPortLabelValues...)
 	}
 }
+
+// collectTotalGPUPower collects the total GPU power consumption from the Control endpoint
+func (c *ChassisCollector) collectTotalGPUPower(ch chan<- prometheus.Metric, chassisID string, logger *slog.Logger) {
+	// Construct the control path for total GPU power
+	// This control may not exist for all chassis types, which is expected
+	controlPath := fmt.Sprintf("/redfish/v1/Chassis/%s/Controls/TotalGPU_Power_0", chassisID)
```
[the remark this answers is not in the document]
Comment (ben-kulp) [reply collector/chassis_collector.go:536]: In gofish, the Chassis has a Controls() method that returns all controls so we should be able to get these dynamically.

**Review by at-ishikawa — APPROVED** [review]
